src/twisted/web/microdom.py

```diff
@@ -422,16 +421,16 @@ def cloneNode(self, deep=0, parent=None):
 
     def writexml(self, stream, indent='', addindent='', newl='', strip=0,
                  nsprefixes={}, namespace=''):
-        stream.write("<![CDATA[")
-        stream.write(self.nodeValue)
-        stream.write("]]>")
+        stream.write(u"<![CDATA[")
+        stream.write(u"" + self.nodeValue)
+        stream.write(u"]]>")
 
 def _genprefix():
     i = 0
     while True:
         yield  'p' + str(i)
         i = i + 1
-genprefix = _genprefix().next
+genprefix = next(_genprefix())
```

Comment:
hmm, i can't see where is that called in `test_xml`, and it would be odd that it wouldn't throw an exception: since currently `genprefix = next(_genprefix())`, that is already a string, so calling `genprefix()` like on the line i linked is of course an error

indeed, the previous version used `genprefix = _genprefix().next` so that `genprefix` was a callable that returned a new, unique string each time it was called

this was because only one `_genprefix()` generator was ever created, and it was repeatedly `.next()`-ed. 
That's why i suggested also only creating one `_genprefix()` generator, and `next()`-ing it repeatedly where needed. This way, subsequent calls to `next(genprefix)` would get different strings, since `genprefix` was the same generator each time

(which is to say, we can't do eg. `prefix = next(_genprefix())` because then we'd get the same string every time)